events/ancient_magic_gain_talent_events.txt

```diff
@@ -42,96 +39,36 @@ ancient_magic_gain_talent_events.002 = {
 ancient_magic_gain_talent_events.003 = {
 	hidden = yes
 	immediate = {
-		if = {
-			limit = {
-				is_ai = no
-			}
-			if = {
-				limit = {
-					has_game_rule = PC_magic_level_1
-				}
-				add_trait = magic_potential_1
-			}
-			if = {
-				limit = {
-					has_game_rule = PC_magic_level_2
-				}
-				add_trait = magic_potential_2
-			}
-			if = {
-				limit = {
-					has_game_rule = PC_magic_level_3
-				}
-				add_trait = magic_potential_3
-			}
-		}
-		if = {
+		give_base_mana_affinity_game_start_effect = yes
+	}
+}
+
+ancient_magic_gain_talent_events.004 = { #Give children mana affinity based on parrents
+	hidden = yes
+	type = empty
+	immediate = {
+		every_living_character = {
 			limit = {
-				is_ai = yes
-			}
-			random = {
-				chance = {
-					value = 0
-					if = {
-						limit = {
-							has_game_rule = NPC_magic_chance_001
-						}
-						add = 1
+				OR = {
+					AND = {
+						exists = mother
```

Comment:
look to line 34 and 42 please, its meant to work exactly like this, in sense that i know that many chatracters dont have parents and thats why this is only called on those that have its correction so that mana affinity is based on their parents,in sense everyone gets mana affinity with this script give_base_mana_affinity_game_start_effect , but i wanted to make sure that those who have parents have their mana affinity based upon them

Reply:
Aha. Good job.